### ISSUE TITLE
Bad content-type returned when downloading file

### ISSUE THREAD

**Issue by GaryTowers** [strongloop/loopback-component-storage#127]
The storage-handler `download` method is returning the file name as a content-type instead of the proper content type. You can see on line [237](https://github.com/strongloop/loopback-component-storage/blob/master/lib/storage-handler.js#L237) and [250](https://github.com/strongloop/loopback-component-storage/blob/master/lib/storage-handler.js#L250). Is there a way to set it manually?

**Comment by superkhau** [comment]
You should be able to pass `file` into download: https://github.com/strongloop/loopback-component-storage/blob/master/lib/storage-service.js#L255

**Comment by GaryTowers** [comment]
Yes, but the `file` parameter is the file name, not its content-type, which is set to the file name on lines [237](https://github.com/strongloop/loopback-component-storage/blob/master/lib/storage-handler.js#L237) and [250](https://github.com/strongloop/loopback-component-storage/blob/master/lib/storage-handler.js#L250). I created a PR https://github.com/strongloop/loopback-component-storage/pull/128 to fix this issue and assign the correct content-type and is working great so far. Could you review it?

**Comment by superkhau** [comment]
@GaryTowers Commented on your PR, we needs some tests to verify the changes. Let's continue the discussion there.

**Comment by thanickel** [comment]
Can somebody merge this fix. Or at least delete the lines 237 and 250, so that we can send the response object with correct content type.

**Comment by superkhau** [comment]
@thanickel We need unit tests to verify the behaviour before merging. Would you like to submit a PR?

**Comment by GaryTowers** [comment]
@superkhau @thanickel, I found a better solution, I will be posting a new PR in a few minutes along with its test case.

**Comment by GaryTowers** [comment]
Alright! I've submitted a new PR with a proper solution.

While doing the test case I figured **it already exists**. Apparently the issue only occurs when the file  includes a `/` in the name for example: **file:** `avatars/user1.png` . This is common for the `AWS` provider which allows adding `/` on the names to create `virtual folders` on the account `bucket`.

The current test succeeds and this fix should not create regression issues.
